src/utils/process-data.js

```diff
@@ -22,6 +21,24 @@ function flattenMarkdownData() {
     };
 }
 
+/**
+ *
+ * @param {Object} value
+ * @param {Array} path
+ * @param {*} [defaultValue]
+ * @return {*}
+ */
+const getDataFromPath = (value, path, defaultValue) => {
+    return path.reduce((acc, v) => {
+        try {
+            acc = acc[v];
+        } catch (e) {
+            return defaultValue;
+        }
+        return acc;
+    }, value);
+};
```

Comment:
@denar90 what should be the result of:
```js
const res = getDataFromPath({
  foo: {
    error: 'hello'
  }
}, ['foo', 'bar'], 'default');
```
Does it work?

What should be the result of:
```js
const res = getDataFromPath({
  foo: {
    error: {
      baz: 'hello'
    }
  }
}, ['foo', 'bar', 'baz'], { baz: 'world' });
```

Also, try not to use try catch, you can use `in` operator to check if property exists.